### ISSUE TITLE
Why there are many holes and inconsistent regions when exporting the point clouds.

### ISSUE THREAD

**Issue by rockywind** [nerfstudio-project/nerfstudio#1363]
Hi,
The training set has 207 images while the eval set has 23 images.
This is the exporting point clouds.
The script is below.
````
ns-export pointcloud --load-config outputs/data_pk2_230_1gpu_ex/nerfacto/2023-02-07_095509/config.yml \
--output-dir exports/pcd_230_10_10_10_6m/ --num-points 6000000 --remove-outliers True --estimate-normals False \
--use-bounding-box True --bounding-box-min -5 -5 -5 --bounding-box-max 5 5 5
````
![image](https://user-images.githubusercontent.com/14838781/217139858-05477ca0-2bf8-4fab-9687-8e97505f6c2e.png)
This is the rendering image in the eval set.
![image](https://user-images.githubusercontent.com/14838781/217139952-c963425e-005e-41bc-89be-553a9c609a1b.png)
This is the train metrics.
![image](https://user-images.githubusercontent.com/14838781/217140341-8927951c-d902-4613-83a8-bb4566cc7b22.png)
This is the eval metrics.
![image](https://user-images.githubusercontent.com/14838781/217140455-858eb946-413f-490c-bca2-c055e953face.png)

**Comment by tancik** [comment]
NeRFs will often "cheat" the geomertry to get good quality color outputs. This means that in regions without texture, the nerf might add a hole and fill in the scene further back. It is easier to tell what is going on by exploring the scene in the viewer. In the viewer are the holes present?

**Comment by rockywind** [comment]
Hi,
I can't connect the viewer.
Can you help me?
This is the script.
````
ns-train nerfacto  --viewer.websocket-port=7002 \
                --pipeline.model.predict-normals False \
                --experiment-name data_pk2_230_1gpu_ex \
                --data data/own_data/data_pk_230_exhau \
                --load_dir outputs/data_pk2_230_1gpu_ex/nerfacto/2023-02-07_095509/nerfstudio_models \
                --viewer.start-train False
````
![image](https://user-images.githubusercontent.com/14838781/217188203-db0afd47-859a-412f-9ad5-948214d537f9.png)

**Comment by tancik** [comment]
Try adding `--viewer.skip-openrelay True` to your command. Also try waiting a minute or two to connect, sometime it take a little time.

**Comment by rockywind** [comment]
Hi, 
I try adding `--viewer.skip-openrelay True`, but still con't connect the viewer.
This is the script.
````
ns-train nerfacto  --viewer.websocket-port=7002 \
                --pipeline.model.predict-normals False \
                --experiment-name data_pk2_230_1gpu_ex \
                --data data/own_data/data_pk_230_exhau \
                --load_dir outputs/data_pk2_230_1gpu_ex/nerfacto/2023-02-07_095509/nerfstudio_models \
                --viewer.start-train False \
                --viewer.skip-openrelay True \
````

**Comment by tancik** [comment]
This is related to https://github.com/nerfstudio-project/nerfstudio/issues/765

**Comment by rockywind** [comment]
Hi, @tancik 
1. What's the meaning of `bounding_box_min` and `bounding_box_max`?
2. What's the meaning of `num-points`? I increase the number of  'num-points', I found that increase the number of points made the effect of 3D reconstruction worse.
3. Why is the number of point clouds not fixed but dynamic? How to set the optimal number of point clouds？
````
ns-export pointcloud --load-config outputs/data_pk2_230_1gpu_ex/nerfacto/2023-02-07_095509/config.yml \
--output-dir exports/pcd_230_10_10_10_6m/ --num-points 6000000 --remove-outliers True --estimate-normals False \
--use-bounding-box True --bounding-box-min -5 -5 -5 --bounding-box-max 5 5 5
````

**Comment by rockywind** [comment]
Hi, @tancik
How can I improve the 3D reconstruction?
This my training script.
````
export CUDA_VISIBLE_DEVICES=0
ns-train nerfacto  --viewer.websocket-port=7009 \
                --machine.num-gpus 1 \
                --experiment-name kitti_00_double_600 \
                --data data/kitti/00_double_200 \
                --vis tensorboard 
````
 This is my 3D reconstruction result.
https://user-images.githubusercontent.com/14838781/219057994-ab1a39f6-d214-4907-bf0e-59cf2ac10b54.mp4
This is my export point cloud script.
````
ns-export pointcloud --load-config outputs/kitti_00_double_600/nerfacto/2023-02-15_100002/config.yml --output-dir exports/pcd_600_kitti_stereo/ --num-points 6000000 --remove-outliers True --estimate-normals False --use-bounding-box True --bounding-box-min -5 -5 -5 --bounding-box-max 5 5 5
````
This is my render result in validation dataset.
![image](https://user-images.githubusercontent.com/14838781/219060960-f3fa70d2-1edc-4165-a7c2-be657f433334.png)

**Comment by 834810269** [comment]
Hi, @rockywind 
I also want to train nerfacto on kitti-like data, I see that you train on kitti dataset and achieve a good render result. 
So I want to kown how do you configure nerfacto to train so effectively on kitti, how to set parameters

**Comment by cta5425** [comment]
> rockywind

@rockywind did you find any answers to yours questions? I am facing a similar issue and will be glad to have your feedback on this

**Comment by cta5425** [comment]
> Hi, @tancik How can I improve the 3D reconstruction? This my training script.
> 
> ```
> export CUDA_VISIBLE_DEVICES=0
> ns-train nerfacto  --viewer.websocket-port=7009 \
>                 --machine.num-gpus 1 \
>                 --experiment-name kitti_00_double_600 \
>                 --data data/kitti/00_double_200 \
>                 --vis tensorboard 
> ```
> 
> This is my 3D reconstruction result. https://user-images.githubusercontent.com/14838781/219057994-ab1a39f6-d214-4907-bf0e-59cf2ac10b54.mp4 This is my export point cloud script.
> 
> ```
> ns-export pointcloud --load-config outputs/kitti_00_double_600/nerfacto/2023-02-15_100002/config.yml --output-dir exports/pcd_600_kitti_stereo/ --num-points 6000000 --remove-outliers True --estimate-normals False --use-bounding-box True --bounding-box-min -5 -5 -5 --bounding-box-max 5 5 5
> ```
> 
> This is my render result in validation dataset. ![image](https://user-images.githubusercontent.com/14838781/219060960-f3fa70d2-1edc-4165-a7c2-be657f433334.png)

@rockywind did you find any answers to yours questions? I am facing a similar issue and will be glad to have your feedback on this

**Comment by cta5425** [comment]
> Hi, @tancik
> 
> 1. What's the meaning of `bounding_box_min` and `bounding_box_max`?
> 2. What's the meaning of `num-points`? I increase the number of  'num-points', I found that increase the number of points made the effect of 3D reconstruction worse.
> 3. Why is the number of point clouds not fixed but dynamic? How to set the optimal number of point clouds？
> 
> ```
> ns-export pointcloud --load-config outputs/data_pk2_230_1gpu_ex/nerfacto/2023-02-07_095509/config.yml \
> --output-dir exports/pcd_230_10_10_10_6m/ --num-points 6000000 --remove-outliers True --estimate-normals False \
> --use-bounding-box True --bounding-box-min -5 -5 -5 --bounding-box-max 5 5 5
> ```

@rockywind did you find any answers to yours questions? I am facing a similar issue and will be glad to have your feedback on this

**Comment by maturk** [comment]
@cta5425
1) bounding box min/max are the coordinates that define a rectangle in 3D space. You only need 2 points to define a bounding box, i.e. the top right corner and then the bottom left corner. 
2) num points refers to how many points are used to make the point cloud. I think this should be self explanatory. Given e.g. num_points == 1000, the algorithm picks 1000 random pixels in the training images, and backprojects points into a point cloud based on the depth estimate from the trained NeRF.
3) Same as above. You can fix the number of points if you want. Note that a NeRF / Gaussian Splatting model does not create perfect geometry i.e. perfect depth maps. There is only a single loss function which is a photometric loss that guides the entire optimization. You are not guranteed to get perfect geometric reconstruction even if your RGB frames look decent. NeRF/ GS is an ill-posed problem by definition.
